Human-readable C source, produced with ctanglex -l critbit.w (part of cwebx on debian).

andreasvc opened the pull request:
While I applaud any effort to thoroughly document code such as done here, it is unfortunate that someone who wants to work on the C code cannot jump right in and work on the code directly, because the CWEB conversion produces a C file without indentation and lots of line number markers. This version is more useful, I believe. Cheers.

andreasvc committed:
Human-readable C source, produced with ctanglex -l critbit.w (part of cwebx on debian).

jgehring commented:
Hi andreasvc, there are already different projects providing a human-readable implementation, e.g. [glk/critbit](https://github.com/glk/critbit) or my own at [jgehring/critbit89](https://github.com/jgehring/critbit89). I'd vote for this repository to stay the same, e.g. to contain the "original" implementation.

andreasvc commented:
glk/critbit is the one this is forked from. With "human-readable" I meant "preserve indentation"; I didn't change the implementation, I only used a switch while extracting the .C file from the CWEB file that preserves the indentation.

andreasvc committed:
critbit tree for fixed length bytestrings

- changed code to work with fixed length bytestrings instead of variable length,
  null-terminated character string.s
- code to print an ascii representation of the tree

andreasvc committed:
mapping of fixed length bytestrings to values

Now works as general key-value store with fixed length bytestrings as keys and values.

andreasvc committed:
correctly update values for existing keys

andreasvc committed:
My conclusion: not competitive with hashtables.
